pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -299,24 +295,64 @@ func (c *EtcdProxyController) syncHandler(key string) error {
 	// attempt processing again later. This could have been caused by a
 	// temporary network failure, or any other transient reason.
 	if err != nil {
-		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		etcdstorage, updateErr := c.updateEtcdStorageStatus(etcdstorage, etcdstoragev1alpha1.EtcdStorageCondition{
```

Comment:
This pattern is going to be really noisy.  How about refactoring the methods to have a method that does the work for the Deploy condition. When it returns here, you update the status once.

Reply:
> This pattern is going to be really noisy. How about refactoring the methods to have a method that does the work for the Deploy condition. When it returns here, you update the status once.

That method could return a status object for you to use, but it still avoids much of the noise.